File: src/models/population/GreedyPopulation.ts
import Population from "./Population";

export default class GreedyPopulation extends Population {
    protected initialize(): void {
        const randomArray = new Array(this.size)
            .fill(0)
            .map(() => ~~(Math.random() * this.tsp.towns.length));

        this.individuals.forEach((individual, index) => {
            individual.makeGreedy(individual.towns[randomArray[index]]);
        });
    }
}

File: zad1/Models/GreedyPopulation.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace zad1.Models
{
    public class GreedyPopulation : Population
    {
        public GreedyPopulation(TSP tsp) : base(tsp.Towns.Count, tsp)
        {
        }

        public GreedyPopulation(int size, TSP tsp) : base(size, tsp)
        {
        }

        protected override void Initialize()
        {
            if (Size == Tsp.Towns.Count) InitializeBest();
            else InitializeRandom();
        }

        private void InitializeRandom()
        {
            var rng = new Random();
            var randomIndicesList = Enumerable.Repeat(0, Size).Select(index => rng.Next(Individuals[0].Towns.Count))
                .ToList();

            Individuals = Individuals.AsParallel()
                .Select((individual, i) => individual.MakeGreedy(individual.Towns[randomIndicesList[i]])).ToList();
        }

        private void InitializeBest()
        {
            Individuals = Individuals.AsParallel()
                .Select((individual, i) => individual.MakeGreedy(individual.Towns[i])).ToList();
        }
    }
}